### PR TITLE
Bookmarks: Add option to show only text in bookmarks bar

### PR DESCRIPTION
**Pull request by innermous**
Implements a feature, requested at #1735 .

### CODE REVIEW

File: src/lib/bookmarks/bookmarkstoolbarbutton.cpp [hunk 1]
```diff
@@ -61,9 +61,24 @@ void BookmarksToolbarButton::setShowOnlyIcon(bool show)
...
 
+bool BookmarksToolbarButton::showOnlyText() const
+{
+    return m_showOnlyText;
+}
+
+void BookmarksToolbarButton::setShowOnlyText(bool show)
+{
+    m_showOnlyText = show;
+    updateGeometry();
```
Comment (nowrep) [inline src/lib/bookmarks/bookmarkstoolbarbutton.cpp]: Isn't update geometry enough? As in `setShowOnlyIcon`

Comment (innermous) [inline src/lib/bookmarks/bookmarkstoolbarbutton.cpp]: Unfortunately, leftmost button in bookmarks panel keeps on showing icon until it is repainted (via hovering). So QPushButton::update() call may be solution for that.

File: src/lib/bookmarks/bookmarkstoolbar.cpp [hunk 2]
```diff
@@ -123,17 +132,17 @@ void BookmarksToolbar::showOnlyIconsChanged(bool state)
     }
 }
 
-void BookmarksToolbar::openBookmarkInNewTab()
```
Comment (nowrep) [inline src/lib/bookmarks/bookmarkstoolbar.cpp]: This doesn't look right, you removed `openBookmarkInNewTab` and `openBookmarkInNewWindow`.

File: src/lib/bookmarks/bookmarkstoolbarbutton.cpp (hunk 1 above)
Comment (nowrep) [inline src/lib/bookmarks/bookmarkstoolbarbutton.cpp]: Alright, so add it also to `setShowOnlyIcon`.

File: src/lib/bookmarks/bookmarkstoolbar.cpp (hunk 2 above)
Comment (innermous) [inline src/lib/bookmarks/bookmarkstoolbar.cpp]: One second, just a mistake.

File: src/lib/bookmarks/bookmarkstoolbarbutton.cpp (hunk 1 above)
Comment (innermous) [inline src/lib/bookmarks/bookmarkstoolbarbutton.cpp]: Done.

File: src/lib/bookmarks/bookmarkstoolbar.cpp [hunk 3]
```diff
@@ -72,10 +73,14 @@ void BookmarksToolbar::contextMenuRequested(const QPoint &pos)
     menu.addSeparator();
     QAction* actDelete = menu.addAction(QIcon::fromTheme("edit-delete"), tr("Delete"));
     menu.addSeparator();
-    QAction* act = menu.addAction(tr("Show Only Icons"));
-    act->setCheckable(true);
-    act->setChecked(m_bookmarks->showOnlyIconsInToolbar());
-    connect(act, SIGNAL(toggled(bool)), m_bookmarks, SLOT(setShowOnlyIconsInToolbar(bool)));
+    m_actShowOnlyIcons = menu.addAction(tr("Show Only Icons"));
```
Comment (nowrep) [inline src/lib/bookmarks/bookmarkstoolbar.cpp]: Last thing, either create the actions in constructor, or null them after closing the menu (as with `m_clickedBookmark`).

File: src/lib/bookmarks/bookmarkstoolbar.cpp [hunk 4]
```diff
@@ -115,6 +120,10 @@ void BookmarksToolbar::bookmarksChanged()
 
 void BookmarksToolbar::showOnlyIconsChanged(bool state)
 {
+    if (state) {
```
Comment (nowrep) [inline src/lib/bookmarks/bookmarkstoolbar.cpp]:
```cpp
if (state && m_actShowOnlyText) {
```

just to be sure

**Comment by nowrep** [comment]
Thanks, next time add `Closes #issuenum` to the commit message to automatically close the issue :)

**Pull request merged**